pandas-stubs/_typing.pyi

```diff
@@ -80,14 +79,10 @@ class WriteExcelBuffer(WriteBuffer[bytes], Protocol):
     def truncate(self, size: Union[int, None] = ...) -> int: ...
 
 FilePath = Union[str, PathLike[str]]
+FilePathOrBuffer = Union[
+    FilePath, IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper, mmap
+]
 
-Buffer = Union[IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper, mmap]
-FileOrBuffer = Union[str, Buffer[AnyStr]]
-FilePathOrBuffer = Union[PathLike[str], FileOrBuffer[AnyStr]]
-FilePathOrBytesBuffer = Union[PathLike[str], WriteBuffer[bytes]]
```

Comment:
All these buffers do not exist in pandas. Only `FilePathOrBytesBuffer` is used in the stubs, so I merged them.